src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name);
+    this.health = health;
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+
+    return this.hidden;
+  }
+
+  apply(sourceHealth, damage) {
```

Comment:
The `apply` method takes `sourceHealth` as a parameter, but it is not used in the method. You can remove this parameter if it's not needed.